mobile-app/src/components/HeaderOptions.tsx

```diff
@@ -207,7 +218,49 @@ export function editProfileOptions(
       <RoundedButton
         text="Done"
         icon={IconCheck}
-        onPress={() => {
+        onPress={async () => {
+          // if (auth.currentUser == null) {
+          //   console.log("Cannot edit profile: User not logged in")
+          //   return;
+          // }
+
+          //TODO Ensure username is unique
+          const usersRef = collection(firestore, "users");
+          const q = query(
+            usersRef,
+            where("username", "==", profileFormData.username),
+          );
+          const queryResults = await getDocs(q);
+
+          // Check if were changing username, if so check if username already exist in DB
+          if (
+            userData.username !== profileFormData.username &&
+            queryResults.size >= 1
```

Comment:
you can do a `!queryResults.empty`. Also call it a `querySnapshot` cuz convention